app/src/main/java/com/calendar/Activity/CountDays.java

```diff
@@ -0,0 +1,153 @@
+                .build();
+        setMaterialDatePickerListener();
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private void setMaterialDatePickerListener() {
+        materialDatePicker.addOnPositiveButtonClickListener(selection -> {
+            dateRangeText.setText(materialDatePicker.getHeaderText());
+            Date startDate = new Date(selection.first);
+            Date endDate = new Date(selection.second);
+            LocalDate startLocalDate = dateToLocalDate(startDate);
+            LocalDate endLocalDate = dateToLocalDate(endDate);
+            countDays(startLocalDate, endLocalDate);
+        });
+    }
+
+    public void selectDateRange(View view) {
+        materialDatePicker.show(getSupportFragmentManager(), "tag_picker");
+    }
+
+    private LocalDate dateToLocalDate(Date date) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        int year = calendar.get(Calendar.YEAR);
+        int month = calendar.get(Calendar.MONTH) + 1;
+        int day = calendar.get(Calendar.DAY_OF_MONTH);
+        return LocalDate.of(year, month, day);
+    }
+
+    public void countDays(LocalDate startDate, LocalDate endDate) {
+        numberOfFreeDays = 0;
+        long numberOfDays = ChronoUnit.DAYS.between(startDate, endDate) + 1;
+
+        databaseReference.addListenerForSingleValueEvent(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                for (LocalDate date = startDate; date.isBefore(endDate.plusDays(1)); date = date.plusDays(1)) {
+                    if (isFree(dataSnapshot, date)) {
+                        numberOfFreeDays++;
+                    }
+                }
+                numberOfFreeDaysText.setText(valueOf(numberOfFreeDays));
+                numberOfWorkingDaysText.setText(valueOf(numberOfDays - numberOfFreeDays));
+            }
+
+            @Override
+            public void onCancelled(@NonNull DatabaseError error) {
+                Log.e(TAG, "calendarClicked - onCancelled", error.toException());
+            }
+        });
+    }
+
+    private boolean isFree(DataSnapshot dataSnapshot, LocalDate date){
+        if (isWeekend(date.getDayOfWeek()) || isHoliday(dataSnapshot,valueOf(date))) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isWeekend(DayOfWeek dayOfWeek){
+        if (dayOfWeek.getValue() == 5 || dayOfWeek.getValue()== 6) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isHoliday(DataSnapshot dataSnapshot, String date) {
+        if (checkAllEventsInFolder(dataSnapshot, date) ||
+                checkAllEventsInFolder(dataSnapshot, convertDateStringToRegardlessOfTheYear(date))) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean checkAllEventsInFolder(DataSnapshot dataSnapshot, String date) {
```

Comment:
It can be simplified.

Reply:
Well in this case I don't see an opportunity to simplify.

Reply:
Right, my fault.